go/gilded-rose_test.go

```diff
@@ -1,15 +1,151 @@
 package main
 
-import "testing"
-
+import (
+	"fmt"
+	"testing"
+	"strconv"
+)
+func Print_Items(items []*Item){
+	fmt.Println("name, sellIn, quality")
+	for i := 0; i < len(items); i++ {
+		fmt.Println(items[i])
+	}
+	fmt.Println("")
+}
 func Test_Foo(t *testing.T) {
 	var items = []*Item{
-		&Item{"foo", 0, 0},
+		&Item{"+5 Dexterity Vest", 10, 20},
+		&Item{"Aged Brie", 2, 0},
+		&Item{"Elixir of the Mongoose", 5, 7},
+		&Item{"Sulfuras, Hand of Ragnaros", 0, 80},
+		&Item{"Sulfuras, Hand of Ragnaros", -1, 80},
+		&Item{"Backstage passes to a TAFKAL80ETC concert", 15, 20},
+		&Item{"Backstage passes to a TAFKAL80ETC concert", 10, 49},
+		&Item{"Backstage passes to a TAFKAL80ETC concert", 5, 49},
+		&Item{"Conjured Mana Cake", 15, 40}, // <-- :O
+		&Item{"Conjured Mana Cake", 3, 46}, // <-- :O
+	}
+
+	days := 12
+
+	for day := 0; day < days; day++ {
+		fmt.Printf("-------- day %d --------\n", day)
+		Print_Items(items)
+		UpdateQuality(items)
+	}
```

Comment:
I would suggest to normally include this as part of the test case, so that like mocking the time or a network query in more complex tests, you can control the entire test and make it check whatever you need it to. Does that make sense?